newsService/src/main/java/com/kovan/app/service/NewsService.java

```diff
@@ -128,8 +127,20 @@ private List<NewsDto.Article> createList(List<NewsDto.Article> list) {
         return (isNull(list)) ? new ArrayList<>() : list;
     }
     private List<NewsDto> fetchSavedNews() {
-        List<String> dates = Arrays.asList(yesterday.toString(), today.toString());
-        List<NewsEntity> newsEntities = newsRepository.findByPublishedAtIn(dates);
+        String todayDate = today.toString();
+        String yesterdayDate = yesterday.toString();
+
+        List<NewsEntity> newsEntities = new ArrayList<>();
+
+        Optional<NewsEntity> todayNews = newsRepository.findByPublishedAt(todayDate);
```

Comment:
check without hitting database whether data is present go to database otherwise not needed.

Reply:
Reduced database hit.